datafusion/functions-aggregate-common/src/aggregate/groups_accumulator/prim_op.rs

```diff
@@ -93,20 +94,27 @@ where
         opt_filter: Option<&BooleanArray>,
         total_num_groups: usize,
     ) -> Result<()> {
```

Comment:
this seems rather small? won't larger values have lower  overhead?

Reply:
> this seems rather small? won't larger values have lower overhead?

Does it mean block size?

Reply:
yes, i mean block size. I would expect something like batchsize (4-8k), or maybe even bigger to have lower overhead? Did you run some experiments?

Reply:
> yes, i mean block size. I would expect something like batchsize (4-8k), or maybe even bigger to have lower overhead? Did you run some experiments?

Yes, I try it.

Now I set `block_size = batch_size`. 

I try the smaller `batch_size` like 1024, and `this pr` show improvement compared to `main`.

It is due to `this pr` can also eliminating the call of `Array::slice`, which is non-trivial due to the computation of null count.

Reply:
Ah I see that batch size is used by default :).
Yeah I think it makes sense to test it a bit further, maybe for this a slightly larger value (e.g. 2x, 4x batch size) will be beneficial when the cardinality is above the batch size.

Also at some point might make sense to think of it in size in memory instead of number of elements (e.g. block of `u8` values might hold 16x more values than u128).

Reply:
I found the hash compute for primitive is actually non-trivial...
And in high cardinality case, the duplicated hash compute led by rehash is really expansive!

I am experimenting about also save hash in hashtable like multi group by. Actually I have found improvement in the new added query in `extended.sql`.

Reply:
> Ah I see that batch size is used by default :). Yeah I think it makes sense to test it a bit further, maybe for this a slightly larger value (e.g. 2x, 4x batch size) will be beneficial when the cardinality is above the batch size.
> 
> Also at some point might make sense to think of it in size in memory instead of number of elements (e.g. block of `u8` values might hold 16x more values than u128).

It is really a good idea to split block by size rather than num rows, I will experiment on it after trying some exist ideas later.